protocol/tx/transaction.go

```diff
@@ -25,25 +25,28 @@ func ComputeOutputID(sc *bc.SpendCommitment) (h bc.Hash, err error) {
 			err = r
 		}
 	}()
-	o := newOutput(valueSource{
-		Ref:      sc.SourceID,
-		Value:    sc.AssetAmount,
-		Position: sc.SourcePosition,
-	}, program{VMVersion: sc.VMVersion, Code: sc.ControlProgram}, sc.RefDataHash, 0)
+	o := newOutput(program{VMVersion: sc.VMVersion, Code: sc.ControlProgram}, sc.RefDataHash, 0)
+	o.setSourceID(sc.SourceID, sc.AssetAmount, sc.SourcePosition)
 
 	h = entryID(o)
 	return h, nil
 }
 
 // TxHashes returns all hashes needed for validation and state updates.
 func TxHashes(oldTx *bc.TxData) (hashes *bc.TxHashes, err error) {
+	defer func() {
```

Comment:
This is a little surprising… I didn't think we would be changing how error handling works in this PR. Is it debugging code?